zebedee-cms/src/main/java/com/github/onsdigital/zebedee/util/versioning/VersionNotFoundException.java

```diff
@@ -24,16 +22,15 @@ public VersionNotFoundException(final Throwable cause) {
         super(cause);
     }
 
-    public static VersionNotFoundException versionsNotFoundException(Collection collection,
-                                                                     List<MissingVersion> missingVersions) {
+    public static VersionNotFoundException versionsNotFoundException(List<MissingVersion> missingVersions) {
         String details = missingVersions
                 .stream()
-                .map(i -> i.toString())
+                .map(i -> i.getCurrentURI())
+                .collect(Collectors.toSet())
+                .stream()
```

Comment:
Why is this being collected as a set and then streamed again? I could only think it was to de-dupe the collection. If so, consider using `.distinct()` to save collecting and streaming twice

Reply:
100%. I was looking for something to de-dupe but couldn't think what it was called. I will make the change now.